tests/Template/ConditionalReturnTypeTest.php

```diff
@@ -859,6 +859,22 @@ function test(string $s): void {
                     }
                     '
             ],
+            'returnTypeBasedOnPhpVersionId' => [
+                '<?php
+                    /**
+                     * @psalm-return (PHP_VERSION_ID is int<70300, max> ? string : int)
```

Comment:
I would never have thought of using int ranges in that way, that's quite creative :thinking: :bulb:.

Reply:
Well, I was _really_ glad to find out that psalm supports ranges, as specifying versions with constants / lists would not be so easy. Ranges aren't quite documented...

That being said, docs on conditional return types don't mention existing special cases of `func_num_args()` and `PHP_MAJOR_VERSION` either.